core/src/main/java/com/bankbazaar/tripmanager/manager/TripActivityManager.java

```diff
@@ -20,11 +21,11 @@ public TripActivity saveTripActivity(TripActivity data)
             return tripActivityRepository.save(data);
         }
         else{
-            Optional<TripActivity> presentData = exists(data.getTripId());
+            Optional<TripActivity> presentData = exists(data.getTripId().getTripId());
             if(presentData.isPresent())
             {
-                TripActivity newData = updateData(presentData.get(), data);
-                return tripActivityRepository.save(newData);
+                updateData(presentData.get(), data);
```

Comment:
Ideally a method should not be updating an input parameter. The update data should create a new object, merge the two data and then send it back.

Also is the update method actually updating data or merging 2 objects? Update should ideally save to db as well

Reply:
here the input parameter is not updated. i update an object of the existing details using the input parameter and update that.